### PR TITLE
[INLONG-8524][Agent] Update the JVM Options for Agent

### LINKED ISSUE #8524: [Improve][Agent] Update the JVM Options for Agent
### Description

Add G1GC and gc-logs related options.

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by dockerzhang**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8524

### Motivation

Unify the JVM options for Agent.

### Modifications

Add G1GC and gc-logs related options.

### CODE REVIEW

**Review by doleyzi — APPROVED** [review]

**Review by justinwwhuang — COMMENTED** [review]
LGTM

**Review by justinwwhuang — APPROVED** [review]

File: inlong-agent/bin/agent-env.sh [hunk 1]
```diff
@@ -37,15 +37,16 @@ else
 fi
 
 if [ -z "$AGENT_JVM_HEAP_OPTS" ]; then
-  HEAP_OPTS="-Xms512m -Xmx6656m"
+  HEAP_OPTS="-Xmx512m -Xss512k"
 else
   HEAP_OPTS="$AGENT_JVM_HEAP_OPTS"
 fi
-GC_OPTS="-XX:SurvivorRatio=6 -XX:+UseMembar -XX:+UseConcMarkSweepGC -XX:+CMSParallelRemarkEnabled -XX:+CMSScavengeBeforeRemark -XX:ParallelCMSThreads=3 -XX:+TieredCompilation -XX:+UseCMSCompactAtFullCollection -verbose:gc -Xloggc:$BASE_DIR/logs/gc.log -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=$BASE_DIR/logs/ -XX:+CMSClassUnloadingEnabled -XX:CMSInitiatingOccupancyFraction=60 -XX:CMSFullGCsBeforeCompaction=1 -Dsun.net.inetaddr.ttl=3 -Dsun.net.inetaddr.negative.ttl=1 -Djava.net.preferIPv4Stack=true"
+GC_OPTS="-XX:+UseG1GC -XX:MaxGCPauseMillis=200 -XX:+TraceClassLoading -XX:InitiatingHeapOccupancyPercent=60 -Djava.net.preferIPv4Stack=true -Dfile.encoding=UTF-8"
```
Comment (aloyszhang) [inline inlong-agent/bin/agent-env.sh:44]: It's better to remove `TraceClassLoading` for the product environment.

**Comment by dockerzhang** [comment]
related to https://github.com/apache/inlong/pull/8147

File: inlong-agent/bin/agent-env.sh (hunk 1 above)
Comment (dockerzhang) [reply inlong-agent/bin/agent-env.sh]: done

**Review by aloyszhang — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**